### PR TITLE
Alex Shaw

### PR DESCRIPTION
**Pull request by alexjshaw**

### CODE REVIEW

File: src/controllers/customer.js [hunk 1]
```diff
@@ -1,53 +1,53 @@
...
-            },
-            // We add an `include` outside of the `data` object to make sure the new contact is returned in the result
-            // This is like doing RETURNING in SQL
-            include: { 
-                contact: true
-            }
-        })
-
-        res.status(201).json({ customer: createdCustomer })
+        const customer = await customerDomain.updateCustomer(name, contact, id)
```
Comment (marc-alzapiedi) [inline src/controllers/customer.js:39]: it looks like you've really modified the legacy code that was here into something much simpler

File: src/controllers/movie.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  try {
+    let movie
+    if (isNaN(Number(id))) {
+      movie = await movieDomain.getMovieByTitle(id)
+    } else {
+      movie = await movieDomain.getMovieByID(id)
+    }
+    res.status(200).json({ movie })
+  } catch (e) {
```
Comment (marc-alzapiedi) [inline src/controllers/movie.js:24]: ok so here it looks like you've taken advantage of the try catch to throw the error

File: src/domains/customer.js
Comment (marc-alzapiedi) [inline src/domains/customer.js:1]: The domains folder you have here is an extra folder that I've never even seen before?

File: src/controllers/movie.js (hunk 2 above)
Comment (alexjshaw) [reply src/controllers/movie.js:24]: Yep, every interaction with the database takes place inside a try catch function.

File: src/controllers/movie.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+  try {
+    const movie = await movieDomain.updateMovie(title, runtimeMins, screenings, id)
+    res.status(201).json({ movie })
+  } catch (e) {
+    if (e instanceof Prisma.PrismaClientKnownRequestError) {
+      if (e.code === "P2002") {
+        return res.status(409).json({ error: "Movie with that title already exists" })
+      }
+      if (e.code === "P2025") {
```
[the remark this answers is not in the document]
Comment (alexjshaw) [reply src/controllers/movie.js:77]: https://www.prisma.io/docs/reference/api-reference/error-reference

Prisma has a whole bunch of inbuilt error codes. "e instanceof Prisma.PrismaClientKnownRequestError" checks whether the error being received is a known prisma code. I then have a couple of specific codes I want to check for, and then if they're found return a 409 or 404 error to the user, and then the 500 error on line 81 is a generic catch all for any error that I haven't written a specific response for.

File: src/controllers/movie.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      error: "Missing fields in request body",
+    });
+  }
+
+  try {
+    const movie = await movieDomain.createMovie(title, runtimeMins, screenings)
+    res.status(201).json({ movie })
+  } catch (e) {
+    if (e instanceof Prisma.PrismaClientKnownRequestError) {
+      if (e.code === "P2002") {
```
Comment (Shylan21) [inline src/controllers/movie.js:50]: What does this part do? Could you briefly go through it please?

File: src/domains/movie.js
Comment (Shylan21) [inline src/domains/movie.js:1]: Wondering how come you have another folder called domain. Did you need it for the extension ?

**Pull request closed without merging**